lib/dal/src/pkg/import.rs

```diff
@@ -1033,17 +1055,62 @@ pub async fn import_only_new_funcs(
+
+    // Iterate through all func specs. If the func is an intrinsic, we need to handle it
+    // separately. If it is any other kind of func, we find or create the func and find or create
+    // its arguments.
     for func_spec in funcs {
-        let func = if func::is_intrinsic(func_spec.name())
-            || SPECIAL_CASE_FUNCS.contains(&func_spec.name())
-        {
-            let func_id = Func::find_id_by_name(ctx, &func_spec.name())
-                .await?
-                .ok_or(PkgError::MissingIntrinsicFunc(func_spec.name().to_owned()))?;
+        if let Some(intrinsic) = IntrinsicFunc::maybe_from_str(func_spec.name()) {
+            if intrinsic == IntrinsicFunc::ResourcePayloadToValue
+                || intrinsic == IntrinsicFunc::NormalizeToArray
+            {
+                if let Some(func_id) =
+                    Func::find_id_by_name_and_kind(ctx, func_spec.name(), FuncKind::Intrinsic)
+                        .await?
+                {
+                    let func = Func::get_by_id_or_error(ctx, func_id).await?;
+                    thing_map.insert(func_spec.unique_id().into(), Thing::Func(func));
+                } else {
+                    let mut override_intrinsic_func_specs =
+                        unsafe_to_install_intrinsic_funcs_pkg.funcs_for_name(intrinsic.name())?;
+                    let override_intrinsic_func_spec = override_intrinsic_func_specs.pop().ok_or(
+                        PkgError::IntrinsicFuncSpecsNoneForName(intrinsic.name().to_owned()),
+                    )?;
+                    if !override_intrinsic_func_specs.is_empty() {
+                        return Err(PkgError::IntrinsicFuncSpecsMultipleForName(
+                            intrinsic.name().to_owned(),
+                        ));
+                    }
+
+                    // Use the override func spec to create the func.
+                    let func = create_func(ctx, &override_intrinsic_func_spec, false).await?;
+
+                    // Find or create the func arguments for the provided spec.
+                    for argument in func_spec.arguments()? {
+                        if FuncArgument::find_by_name_for_func(ctx, argument.name(), func.id)
+                            .await?
+                            .is_none()
+                        {
+                            create_func_argument(ctx, func.id, &argument).await?;
+                        }
+                    }
 
-            Func::get_by_id_or_error(ctx, func_id).await?
+                    // However, use the _provided_ func spec when inserting into the thing map. Why
+                    // not use the override? Because we are initializing the thing map here and the
+                    // users of the thing map don't need the "overrides" re-direction: they just
+                    // have the func they need here.
+                    thing_map.insert(func_spec.unique_id().into(), Thing::Func(func));
```

Comment:
This is the fix to the bug causing the revert. The attached screenshot explains the old behavior and the new behavior.

<img width="677" alt="Screenshot 2025-02-14 at 3 57 57 PM" src="https://github.com/user-attachments/assets/1786191a-678e-45a6-ada6-ed7cb4d4ef5b" />